internal/dao/dp.go

```diff
@@ -211,6 +211,42 @@ func (d *Deployment) Scan(ctx context.Context, gvr, fqn string, wait bool) (Refs
 	return refs, nil
 }
 
+func (d *Deployment) GetPodSpec(path string) (*v1.PodSpec, error) {
+	dp, err := d.Load(d.Factory, path)
+	if err != nil {
+		return nil, err
+	}
+	podSpec := dp.Spec.Template.Spec
+	return &podSpec, nil
+}
+
+func (d *Deployment) SetImages(ctx context.Context, path string, containersPatch map[string]string, initContainersPatch map[string]string) error {
```

Comment:
@Ameausoone Thank you for making the changes. I do think we have a problem here. To issue the patch on the resource we need to track and preserve the containers order especially for the init containers.  Maps in go are unordered so this will not preserve the original pod spec. Here is what I think we should do: Define a custom collection data structure as such:

```go
type ImageSpecs []ImageSpec
type ImageSpec struct {
  index int
  name, dockerImage string
  init bool
}
```

From there you can define methods on ImageSpecs to set and extract the order and the images for init and regular containers based on the input form and apply those during the patch. This will enable us to track the updates in a single collection and thus simplify the signatures.

NOTE: I'll let you make the call on whether you want to tackle this or we could merge and make the changes accordingly

Reply:
Of course ! You talked about it on first PR, sorry I should have see this. I'll fix this.